cmd/nvidia-dra-controller/imex.go

```diff
@@ -99,29 +103,34 @@ func StartIMEXManager(ctx context.Context, config *Config) (*ImexManager, error)
 
 // manageResourceSlices reacts to added and removed IMEX domains and triggers the creation / removal of resource slices accordingly.
 func (m *ImexManager) manageResourceSlices(ctx context.Context, owner resourceslice.Owner, addedDomainsCh <-chan string, removedDomainsCh <-chan string) error {
-	driverResources := resourceslice.DriverResources{}
-	controller, err := resourceslice.StartController(ctx, m.clientset, DriverName, owner, &driverResources)
+	driverResources := &resourceslice.DriverResources{}
+	controller, err := resourceslice.StartController(ctx, m.clientset, DriverName, owner, driverResources)
 	if err != nil {
 		return fmt.Errorf("error starting resource slice controller: %w", err)
 	}
 
+	ico := new(imexDomainOffsets)
 	m.waitGroup.Add(1)
 	go func() {
 		defer m.waitGroup.Done()
 		for {
 			select {
 			case addedDomain := <-addedDomainsCh:
+				offset, err := ico.getOffset(addedDomain)
```

Comment:
Let's call this `add` to mimic the `remove` below
```suggestion
				offset, err := imexDomainOffsets.add(addedDomain)
```